### PR TITLE
refactor: make the channel in the matchspec struct an actual channel

### PR DESCRIPTION
**Pull request by ruben-arts**
This should really help make installation more correct. 

I got the issue that `conda-forge::python` and `conda::python` got the same result after installation. 
So I changed the type of the channel to be a channel from the start. 

WDYT?

### CODE REVIEW

**Review by baszalmstra — COMMENTED** [review]
If you turn it into an Arc<Channel> this looks good to me!

**Comment by ruben-arts** [comment]
> If you turn it into an Arc this looks good to me!

Done, please check if that is correct.

File: crates/rattler_conda_types/src/channel/mod.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ impl Default for ChannelConfig {
 }
 
 /// `Channel`s are the primary source of package information.
-#[derive(Debug, Clone, Serialize, Eq, PartialEq, Hash)]
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Hash)]
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/channel/mod.rs:41]: This is no longer required right? Since you have a custom deserializer?

Comment (ruben-arts) [reply crates/rattler_conda_types/src/channel/mod.rs:41]: Nup

Comment (ruben-arts) [reply crates/rattler_conda_types/src/channel/mod.rs]: Removed

**Pull request merged**